enclave/core/CMakeLists.txt

```diff
@@ -1,96 +1,123 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-if (UNIX)
-set(PLATFORM_SRC
-    linux/reloc.c
-    )
-elseif (WIN32)
-set(PLATFORM_SRC
-    windows/reloc.c
-    )
+set(PLATFORM_SRC "")
+
+if (OE_SGX)
+    list(APPEND PLATFORM_SRC
+        sgx/atexit.c
+        sgx/backtrace.c
+        sgx/calls.c
+        sgx/tracee.c
+        sgx/cpuid.c
+        sgx/debugmalloc.c
+        sgx/entropy.c
+        sgx/exception.c
+        sgx/globals.c
+        sgx/hostcalls.c
+        sgx/init.c
+        sgx/jump.c
+        sgx/keys.c
+        sgx/malloc.c
+        sgx/memory.c
+        sgx/once.c
+        sgx/properties.c
+        sgx/report.c
+        sgx/sbrk.c
+        sgx/spinlock.c
+        sgx/td.c
+        sgx/thread.c
+        sgx/enter.S
+        sgx/exit.S
+        sgx/getkey.S)
+
+    # OS specific sources for SGX.
+    if (UNIX OR USE_CLANGW)
+        list(APPEND PLATFORM_SRC
+            ../../common/sgx/rand.S
+            sgx/linux/reloc.c
+            sgx/linux/threadlocal.c)
+    elseif (WIN32)
+        list(APPEND PLATFORM_SRC
+            ../../common/sgx/rand.asm
+            sgx/windows/reloc.c)
+    endif()
+elseif(OE_TRUSTZONE)
```

Comment:
Shouldn't this be OE_USE_OPTEE not OE_TRUSTZONE?
After all, if you're using TZ with something other than OP-TEE, this section would be wrong